packages/core/src/common/event.ts

```diff
@@ -331,12 +333,12 @@ export namespace WaitUntilEvent {
             delete asyncEvent['waitUntil'];
         }
         if (!waitables.length) {
-            return;
+            return [];
         }
         if (timeout !== undefined) {
-            await Promise.race([Promise.all(waitables), new Promise(resolve => setTimeout(resolve, timeout))]);
+            return Promise.race([Promise.all(waitables), new Promise<undefined>(resolve => setTimeout(resolve, timeout, undefined))]);
```

Comment:
Why do we need to return anything here? I did not find that these API changes are necessary in the PR.

Reply:
It will allow to return things like `WorkspaceEdits` in the future, but I can revert...

Reply:
Could you point me please to place where we will have to use it? It does not seem to be relevant to the issue, but maybe we can extract it to own commit.

Reply:
I would prefer to revert it for now and look at it when we work on the corresponding task. Looking at VS Code they don't have such APIs are still capable to implement everything.